pkg/controller/servicebindingrequest/annotations/configmap_test.go

```diff
@@ -0,0 +1,122 @@
+package annotations
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+
+	"github.com/redhat-developer/service-binding-operator/pkg/testutils"
+	"github.com/redhat-developer/service-binding-operator/test/mocks"
+)
+
+func TestConfigMapHandler(t *testing.T) {
+	type args struct {
+		name      string
+		value     string
+		service   map[string]interface{}
+		resources []runtime.Object
+		expected  map[string]interface{}
+	}
+
+	assertHandler := func(args args) func(*testing.T) {
+		return func(t *testing.T) {
+			f := mocks.NewFake(t, "test")
+
+			for _, r := range args.resources {
+				f.AddMockResource(r)
+			}
+
+			restMapper := testutils.BuildTestRESTMapper()
+
+			bindingInfo, err := NewBindingInfo(args.name, args.value)
+			require.NoError(t, err)
+			handler, err := NewConfigMapHandler(
+				f.FakeDynClient(),
+				bindingInfo,
+				unstructured.Unstructured{Object: args.service},
+				restMapper,
+			)
+			require.NoError(t, err)
+			got, err := handler.Handle()
+			require.NoError(t, err)
+			require.NotNil(t, got)
+			require.Equal(t, args.expected, got.Object)
+		}
+	}
+
+	t.Run("configmap/scalar", assertHandler(args{
+		name:  "servicebindingoperator.redhat.io/status.dbCredentials-password",
+		value: "binding:env:object:configmap",
+		service: map[string]interface{}{
+			"metadata": map[string]interface{}{
+				"namespace": "the-namespace",
+			},
+			"status": map[string]interface{}{
+				"dbCredentials": "the-secret-resource-name",
+			},
+		},
+		resources: []runtime.Object{
+			&corev1.ConfigMap{
+				TypeMeta: metav1.TypeMeta{
+					Kind: "ConfigMap",
+				},
+				ObjectMeta: metav1.ObjectMeta{
+					Namespace: "the-namespace",
+					Name:      "the-secret-resource-name",
+				},
+
+				Data: map[string]string{
+					"password": "hunter2",
+				},
+			},
+		},
+		expected: map[string]interface{}{
+			"configmap": map[string]interface{}{
+				"password": "hunter2",
+			},
+		},
+	}))
+
+	t.Run("configmap/map", assertHandler(args{
+		name:  "servicebindingoperator.redhat.io/status.dbCredentials",
+		value: "binding:env:object:configmap",
+		service: map[string]interface{}{
+			"metadata": map[string]interface{}{
+				"namespace": "the-namespace",
+			},
+			"status": map[string]interface{}{
+				"dbCredentials": "the-secret-resource-name",
+			},
+		},
+		resources: []runtime.Object{
+			&corev1.ConfigMap{
+				TypeMeta: metav1.TypeMeta{
+					Kind: "ConfigMap",
+				},
+				ObjectMeta: metav1.ObjectMeta{
+					Namespace: "the-namespace",
+					Name:      "the-secret-resource-name",
+				},
+				Data: map[string]string{
+					"password": "hunter2",
+					"username": "AzureDiamond",
+				},
+			},
+		},
+		expected: map[string]interface{}{
```

Comment:
In test `configmap/scalar`, I can see  expected is `configmap->pasword` because there is a single item at second level.

In test `configmap/map`, I can see expected is `configmap->status->dbCredentials`.

Why not keep things consistent and have `configmap->status->dbCredentials->password` for `configmap/scalar as well?`

Reply:
In `configmap/map`, the item being extracted iss `servicebindingoperator.redhat.io/status.dbCredentials`: see there's no `-password` in this value, indicating to extract the whole node from the service resource.

The other test case states `servicebindingoperator.redhat.io/status.dbCredentials-password`, indicating (from the examples and usage at hand) indicating that the `password` field should be extracted from `status.dbCredentials` and aliased to `password`.

Aliases are used with annotations consuming data from ConfigMap and Secret, and doesn't have any usage with self-referential data extraction (when the data is extracted from the service resource directly and not from an external resource).

Reply:
Thanks for the explaination